src/open_samus_returns_rando/files/templates/randomizerpowerup.lua

```diff
@@ -96,6 +97,34 @@ function RandomizerPowerup.MarkLocationCollected(actorOrName)
     Blackboard.SetProp(playerSection, propName, "b", true)
 end
 
+local AeionAbilityClouds = {
+    {"s000_surface", "LE_SpecialAbility_ScanningPulse"},
+    {"s028_area2c", "LE_SpecialAbility_EnergyShield"},
+    {"s030_area3", "LE_SpecialAbility_EnergyWave"},
+    {"s060_area6", "LE_SpecialAbility_PhaseDisplacement"},
+}
+
+function RandomizerPowerup.ActivateSpecialEnergy(actorOrName)
+    local name = actorOrName.sName
+    local cloud = "TG_SpecialEnergyCloud"
+    local trigger = cloud .. string.sub(name, 11)
+
+    -- Powerups
+    if string.sub(name, 0, 8) == "LE_Power" and trigger ~= nil then
```

Comment:
Trigger can never be `nil`

Reply:
It's definitely can be. Not every powerup has a cloud, so this filters those out before the call happens.

Reply:
But the lines directly above doing this:
```lua
    local cloud = "TG_SpecialEnergyCloud"
    local trigger = cloud .. string.sub(name, 11)
```
`trigger` can never be `nil` because it will always contain at least "TG_SpecialEnergyCloud"

Reply:
Oh then it should probably be `.... and Game.GetEntity(trigger) ~= nil)`